File: 10816/10816.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10816.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/05/14 22:47:47 by jihoolee          #+#    #+#              #
#    Updated: 2022/05/14 23:40:03 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main():
    _ = input()
    cards_lst = list(map(int, input().split()))
    _ = input()
    test_cases_lst = list(map(int, input().split()))

    cards_dict = dict()
    result = ""

    for num in cards_lst:
        cards_dict[num] = cards_dict.get(num, 0) + 1
    for test_case in test_cases_lst:
        result += str(cards_dict.get(test_case, 0)) + ' '
    print(result)

if (__name__ == '__main__'):
    main()

File: 10816/10816.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10816.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/05/14 22:23:49 by jihoolee          #+#    #+#             */
/*   Updated: 2022/05/14 22:46:23 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <map>

int get_card_count(int card_num, std::map<int, int>& cards) {
  std::map<int, int>::iterator it = cards.find(card_num);

  if (it == cards.end())
    return 0;
  return it->second;
}

void check_cards(std::map<int, int>& cards) {
  int case_len;
  int num_input;

  std::cin >> case_len;
  for (int i = 0; i < case_len; ++i) {
    std::cin >> num_input;
    std::cout << get_card_count(num_input, cards) << ' ';
  }
}

void get_card_input(std::map<int, int>& cards) {
  int cards_count;
  int num_input;

  std::cin >> cards_count;
  for (int i = 0; i < cards_count; ++i) {
    std::cin >> num_input;
    if (cards.count(num_input))
      ++cards[num_input];
    else
      cards.insert(std::map<int, int>::value_type(num_input, 1));
  }
}

int main(void) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(nullptr);
  std::cout.tie(nullptr);

  std::map<int, int>  cards;

  get_card_input(cards);
  check_cards(cards);
  return 0;
}
